### PR TITLE
Upgrade Segger SystemView to V3.54

### PR DESCRIPTION
**Pull request by XuNeo**
## Summary

Upgrade to the latest version V3.54.

Segger official github repo is out-of-date. Downloading from segger website is available but not stable. Thus I mirrored them to https://github.com/XuNeo/SystemView

## Impact
Should have no impact on existing projects.

## Testing
Tested on internal project, and stm32f429i-disco with newly added defconfig.

### CODE REVIEW

**Comment by anchao** [comment]
why not get the source code from segger?
https://www.segger.com/downloads/systemview/SystemView_Src_V354.zip

**Comment by anchao** [comment]
CMake should also needs to be updated appropriately
https://github.com/apache/nuttx/blob/master/drivers/segger/CMakeLists.txt

**Comment by acassis** [comment]
@XuNeo also I think it is not a good idea to download these files from your personal repository. I know did it in good intention, but please don't do it. It is better to use the repository from official company.

**Comment by XuNeo** [comment]
> why not get the source code from segger?

As mentioned above, I originally tested locally and found out official website is not stable. It may be related to many download request in short time.
But I tried today, and it seems works after many rebuild. Download URL is now updated.

> CMake should also needs to be updated appropriately

Done.

**Comment by XuNeo** [comment]
> Error: segger/SystemView/SEGGER/SEGGER_SYSVIEW.c:2509:34: error: cast from pointer to integer of different size [-Werror=pointer-to-int-cast]
 2509 |   ENCODE_U32(pPayload, SHRINK_ID((U32)pHeap));

`sim:segger` reports warning about pointer to integer with different size. Segger may not work with 64bit. Warning is not suppressed.

File: drivers/segger/CMakeLists.txt [hunk 1]
```diff
@@ -82,46 +79,20 @@ if(CONFIG_SEGGER_RTT)
...
+  target_compile_options(SystemView PRIVATE -Wno-shadow
+                                            -Wno-pointer-to-int-cast)
   target_sources(SystemView PRIVATE note_sysview.c
-                                    SystemView/SYSVIEW/SEGGER_SYSVIEW.c)
+                                    SystemView/SEGGER/SEGGER_SYSVIEW.c)
 
-  target_include_directories(
-    SystemView PRIVATE ../../sched config RTT/RTT SystemView/SEGGER
-                       SystemView/SYSVIEW)
+  target_include_directories(SystemView PRIVATE ../../sched config RTT/RTT
```
Comment (anchao) [inline drivers/segger/CMakeLists.txt:95]:
```suggestion
  target_include_directories(SystemView PRIVATE ${NUTTX_DIR}/sched config RTT/RTT
```

**Review by anchao — CHANGES_REQUESTED** [review]

File: drivers/segger/CMakeLists.txt (hunk 1 above)
Comment (XuNeo) [reply drivers/segger/CMakeLists.txt]: Done

**Review by anchao — APPROVED** [review]

**Review by jerpelea — APPROVED** [review]

**Comment by acassis** [comment]
> > why not get the source code from segger?
> 
> As mentioned above, I originally tested locally and found out official website is not stable. It may be related to many download request in short time. But I tried today, and it seems works after many rebuild. Download URL is now updated.
> 
> > CMake should also needs to be updated appropriately
> 
> Done.

@XuNeo you can use our mirror repository to save a backup: https://github.com/NuttX/apps-mirror-pkgs

**Comment by XuNeo** [comment]
> @XuNeo you can use our mirror repository to save a backup: https://github.com/NuttX/apps-mirror-pkgs

@acassis There's no use case in that mirror repository right now. Would you like to setup an example?

**Comment by acassis** [comment]
> > @XuNeo you can use our mirror repository to save a backup: https://github.com/NuttX/apps-mirror-pkgs
> 
> @acassis There's no use case in that mirror repository right now. Would you like to setup an example?

Right, I will create an example for some app that you can yes as reference

**Pull request merged**